### PR TITLE
feature: Auth token forwarding

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Please provide a description of what your change does and why it is needed.

Closes SAP/cloud-sdk-backlog#286.

Enables the `forwardAuthToken` on the enviorment destinations as the [approuter](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/3cc788ebc00e40a091505c6b3fa485e7.html) also allows it.

### CODE REVIEW

File: packages/core/src/connectivity/scp-cf/destination/destination-from-env.ts [hunk 1]
```diff
@@ -161,3 +167,26 @@ export function searchEnvVariablesForDestination(
...
+): [DestinationAuthToken] | undefined {
+  if (token) {
+    const decoded = decodeJwt(token);
+    logger.info(
+      "Option 'forwardAuthToken' enabled on destination. Using the initial token for the destination."
+    );
+    return [
+      {
+        value: token,
+        expiresIn: decoded.exp!.toString(),
```
Comment (jjtang1985) [inline packages/core/src/connectivity/scp-cf/destination/destination-from-env.ts:182]: If the service token expires, will you get a refreshed token automatically?

Comment (FrankEssenberger) [reply packages/core/src/connectivity/scp-cf/destination/destination-from-env.ts:182]: no, The token we are talking about is the initial one from the incoming request. I would assume that his token is still valid.

**Review by jjtang1985 — APPROVED** [review]
LGTM.
Please consider updating the documentation.

**Comment by FrankEssenberger** [comment]
Here is the PR for the documentation change: https://github.com/SAP/cloud-sdk/pull/476

**Pull request merged**